### DISCUSSION TITLE
Studio query with large custom function

### DISCUSSION THREAD

**Discussion by andrewdavey** [ravendb/ravendb#12216] [Q&A]
In Studio (v5.0.9) I'm trying write a query that declares a custom javascript function. However, the function is so long that it's breaking the HTTP request when trying to run the query.

Chrome dev tools report this:

```
Failed to load resource: the server responded with a status of 414 (URI Too Long)
```

Is there a way to make it post the query, instead of relying on HTTP get?

**Comment by ml054** [comment]
Hi, 

I've opened issue for that: https://issues.hibernatingrhinos.com/issue/RavenDB-16744

@andrewdavey please also keep in mind that having large javascript block in query might have influence on query performance, so it is advised to reduce query size or move logic to index to avoid computing that on every call. 

Thanks for sharing your feeback
